http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -618,71 +616,114 @@ class LeoPubsubMessageSubscriber[F[_]: Timer: ContextShift](
+      //  4. create and poll app
+      //
+      // Numbers 1-3 are all Google calls; (4) is a helm call. If either of (1) or (2) are
+      // necessary then we will do the _initial_ GKE call synchronous to the pubsub processing so
+      // we can nack the message on errors. Monitoring all creations will be asynchronous,
+      // and (3) and (4) will always be asynchronous.
+
+      // Create the cluster synchronously
+      createClusterResultOpt <- msg.cluster
+        .traverse { c =>
+          gkeInterp.createCluster(CreateClusterParams(c.clusterId, msg.project, List(c.defaultNodepoolId), false))
+        }
+        .adaptError {
           case e =>
             PubsubKubernetesError(
-              AppError(e.getMessage(), ctx.now, ErrorAction.CreateGalaxyApp, ErrorSource.Cluster, None),
+              AppError(e.getMessage, ctx.now, ErrorAction.CreateGalaxyApp, ErrorSource.Cluster, None),
               Some(msg.appId),
               false,
               msg.cluster.map(_.defaultNodepoolId),
               msg.cluster.map(_.clusterId)
             )
         }
 
-        _ <- msg.nodepoolId.traverse { nodepoolId =>
-          gkeInterp.createAndPollNodepool(nodepoolId, msg.appId).adaptError {
+      // Create the nodepool synchronously if we didn't need to create the cluster
+      createNodepoolResultOpt <- if (createClusterResultOpt.isDefined) F.pure(none[CreateNodepoolResult])
+      else
+        msg.nodepoolId
+          .traverse(nodepoolId => gkeInterp.createNodepool(CreateNodepoolParams(nodepoolId, msg.project)))
+          .adaptError {
             case e =>
               PubsubKubernetesError(
-                AppError(e.getMessage(), ctx.now, ErrorAction.CreateGalaxyApp, ErrorSource.Nodepool, None),
+                AppError(e.getMessage, ctx.now, ErrorAction.CreateGalaxyApp, ErrorSource.Nodepool, None),
                 Some(msg.appId),
                 false,
-                Some(nodepoolId),
+                msg.nodepoolId,
                 None
               )
           }
+
+      // build asynchronous task
+      task = for {
+        // create disk if necessary
+        _ <- createDiskForApp(msg).adaptError {
+          case e =>
+            PubsubKubernetesError(
+              AppError(e.getMessage, ctx.now, ErrorAction.CreateGalaxyApp, ErrorSource.Disk, None),
+              Some(msg.appId),
+              false,
+              None,
+              None
+            )
         }
-      } yield ()
 
-      diskCreation = createDiskForApp(msg).adaptError {
-        case e =>
-          PubsubKubernetesError(
-            AppError(e.getMessage(), ctx.now, ErrorAction.CreateGalaxyApp, ErrorSource.Disk, None),
-            Some(msg.appId),
-            false,
-            None,
-            None
-          )
-      }
-      parPreAppCreationSetup = List(createClusterAndNodepool, diskCreation).parSequence.flatMap(_ => F.unit)
+        // monitor cluster creation if necessary
```

Comment:
previously we did the cluster monitoring and nodepool handling in parallel to disk creation and monitoring via `parPreAppCreationSetup = List(createClusterAndNodepool, diskCreation).parSequence.flatMap(_ => F.unit)`. Not sure how long disk creation/monitoring takes, but it seems good to not hold up other processing on that

Reply:
Good point, previously I thought monitoring this disk creation was async anyway so it didn't need to be parallel.. but now I see that this disk creation+monitoring is synchronous. So maybe it makes sense to do it in parallel to create cluster+nodepool.

Reply:
Just added back parallel disk creation/monitoring